src/main/scala/cromwell/engine/CallActor.scala

```diff
@@ -187,7 +187,8 @@ class CallActor(key: CallKey, locallyQualifiedInputs: CallInputs, backend: Backe
     )
 
     val message = executionResult match {
-      case SuccessfulExecution(outputs, returnCode, hash) => WorkflowActor.CallCompleted(key, outputs, returnCode, hash)
+      case SuccessfulExecution(outputs, returnCode, hash) =>
+        WorkflowActor.CallCompleted(key, outputs, returnCode, if (workflowDescriptor.writeToCache) hash else "")
```

Comment:
Yeah, it's horrible knowing that you can't put a firstName: String into a lastName: String accidentally.

And on the latter one, yeah the point is to not heavily overload these things anyways, if one were going down this path and it's not really a SymbolHash it shouldn't use SymbolHash. Otherwise you're right back where you started